### PR TITLE
update gitignore

### LINKED ISSUE #3505: update gitignore
**What would you like to be added**:
Ignore vendor directory

**Why is this needed**:

### PR DESCRIPTION
**Pull request by zouyee**
Signed-off-by: zouyee <zounengren@cmss.chinamobile.com>

### 1. Why is this pull request needed and what does it do?
Ignore vendor directory

### 2. Which issues (if any) are related?
fix https://github.com/coredns/coredns/issues/3505
### 3. Which documentation changes (if any) need to be made?
NONE
### 4. Does this introduce a backward incompatible change or deprecation?
NONE

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -9,9 +9,38 @@ go-test-tmpfile*
 coverage.txt
 .idea/
 .DS_Store
+build/
+release/
```
Comment (miekg) [inline .gitignore]: only these 2 additions make sense to me, the rest was never there and this seems not to have caused problems, so please remove those.

File: go.sum [hunk 2]
```diff
@@ -68,10 +68,6 @@ github.com/aliyun/aliyun-oss-go-sdk v0.0.0-20190307165228-86c17b95fcd5/go.mod h1
 github.com/apache/thrift v0.12.0 h1:pODnxUFNcjP9UTLZGTdeh+j16A8lJbRvD3rOtrk/7bs=
 github.com/apache/thrift v0.12.0/go.mod h1:cp2SuWMxlEZw2r+iP2GNCdIi4C1qmUzdZFSVb+bacwQ=
 github.com/aws/aws-sdk-go v1.23.0/go.mod h1:KmX6BPdI08NWTb3/sm4ZGu5ShLoqVDhKgpiN924inxo=
-github.com/aws/aws-sdk-go v1.25.36 h1:4+TL/Y2G5hsR1zdfHmjNG1ou1WEqsSWk8v7m1GaDKyo=
```
Comment (miekg) [inline go.sum]: this doesn't belong in this pr - please remove. We *do* need to figure out how to perform 'go mod tidy' from time to time.

**Comment by miekg** [comment]
ping here, otherwise I'll close. Thanks

**Comment by zouyee** [comment]
update：Ignore vendor directory
